core/types/crosslink.go

```diff
@@ -13,27 +13,38 @@ import (
 
 // CrossLink is only used on beacon chain to store the hash links from other shards
 // signature and bitmap correspond to |blockNumber|parentHash| byte array
-// Captial to enable rlp encoding
+// Capital to enable rlp encoding
 // Here we replace header to signatures only, the basic assumption is the committee will not be
 // corrupted during one epoch, which is the same as consensus assumption
 type CrossLink struct {
 	HashF        common.Hash
 	BlockNumberF *big.Int
+	ViewIDF      *big.Int
 	SignatureF   [96]byte //aggregated signature
 	BitmapF      []byte   //corresponding bitmap mask for agg signature
 	ShardIDF     uint32   //will be verified with signature on |blockNumber|blockHash| is correct
 	EpochF       *big.Int
 }
 
 // NewCrossLink returns a new cross link object
-// epoch is the parentHeader's epoch
-func NewCrossLink(header *block.Header, epoch *big.Int) CrossLink {
+func NewCrossLink(header *block.Header, parentHeader *block.Header) CrossLink {
```

Comment:
New implies returns pointer , make it a pointer and all these crosslink methods should have pointer receiver

Reply:
Sure thing, changed in latest